src/flag_gems/ops/unique.py

```diff
@@ -0,0 +1,846 @@
+    tile_sum = torch.empty(
+        (global_num_ctas,), dtype=torch.int32, device=sorted_data.device
+    )
+    data_out = torch.empty_like(sorted_data)
+    inverse_indices = torch.empty_like(sorted_data, dtype=torch.int32)
+    idx = None
+    if return_counts:
+        idx = torch.empty_like(inverse_indices)
+
+    # launch kernel
+    with torch.cuda.device(sorted_data.device.index):
+        local_ne_flat_kernel[grid](
+            sorted_data,  # in
+            ne_result,
+            tile_sum,  # out
+            global_num_ctas,
+            num_tasks,
+            tiles_per_cta=tiles_per_cta,
+            tile_size=tile_size,
+            one_tile_per_cta=tiles_per_cta == 1,
+            num_warps=num_warps,
+        )
+        global_cumsum_flat_kernel[grid](
+            ne_result,
+            tile_sum,  # in
+            sorted_data,
+            sorted_indices,  # in
+            data_out,
+            inverse_indices,
+            idx,  # out
+            num_ctas,
+            global_num_ctas,
+            next_power_global_num_ctas,
+            num_tasks,
+            tiles_per_cta=tiles_per_cta,
+            tile_size=tile_size,
+            one_tile_per_cta=tiles_per_cta == 1,
+            return_counts=return_counts,
+            num_warps=num_warps,
+        )
+        out_size = tile_sum[-1].item() + 1
+        counts = None
+        if return_counts:
+            idx = idx[:out_size]
+            counts = torch.empty_like(idx)
+            output_counts_flat_kernel[grid](
+                idx,
+                num_tasks,  # in
+                counts,  # out
+                out_size,
+                tiles_per_cta,
+                tile_size,
+                one_tile_per_cta=tiles_per_cta == 1,
+                num_warps=num_warps,
+            )
+
+    return data_out[:out_size], inverse_indices, counts
+
+
+def simple_unique_flat(
+    sorted_data: torch.Tensor,
+    sorted_indices: torch.Tensor,
+    return_inverse: bool,
+    return_counts: bool,
+):
+    num_tasks = sorted_data.numel()
+    grid = (1, 1, 1)
+
+    # allocate tensor
+    data_out = torch.empty_like(sorted_data)
+    if return_inverse:
+        inverse_indices = torch.empty_like(sorted_data, dtype=torch.int32)
+    else:
+        inverse_indices = None
+    if return_counts:
+        idx = torch.empty_like(sorted_data, dtype=torch.int32)
+    else:
+        idx = None
+    unique_size = torch.empty([1], dtype=torch.int32, device=sorted_data.device)
+
+    # launch kernel
+    with torch.cuda.device(sorted_data.device.index):
+        simple_unique_flat_kernel[grid](
+            sorted_data,
+            sorted_indices,  # in
+            data_out,
+            inverse_indices,
+            idx,
+            unique_size,  # out
+            return_inverse,
+            return_counts,
+            num_tasks,
+            tile_size=triton.next_power_of_2(num_tasks),
+            num_warps=8,
+        )
+    out_size = unique_size.item() + 1
```

Comment:
Pulling data back to CPU could be avoided by passing the result tensor as parameter to a tensor constructor.